0000-interledger-payment-request-3/0000-interledger-payment-request-3.md

```diff
@@ -0,0 +1,113 @@
+
+### Scope
+
+This document defines a suggested format for representing payment requests (as ILP packets) and an approach for receiver-generated conditions. Protocols built on top of IPR MAY use the format provided here, though they can use any format they choose instead.
+
+While this document also provides a recommendation for generating conditions, conditions are often opaque to all parties aside from the receiver. Receivers MAY use any condition they choose, however this specification includes best practices that receivers SHOULD take into consideration.
+
+This document does **not** specify how payment requests are communicated from a receiver to a sender. A number of use case options exist, some of which are described in the [Loopback Transport Protocol - IL RFC 29]().
+
+### Model of Operation
+
+The assumption, when creating an IPR, is that the receiver is requesting a payment from the sender. This can be modelled as an account between the sender and receiver where the balance is currently in favour of the receiver by at least the amount they wish to receive.
+
+Therefor, just like the accounts between any two ILP peers, this balance can be adjusted by the successful exchange of an ILP packet between the peers. To that end, the payment request from the receiver takes the form of an `InterledgerPrepare` packet for a payment from the receiver to the sender which will settle the account between them if it is fulfilled.
+
+The sender can then forward this packet back to the receiver via the Interledger network and, in so doing, get back the fulfillment of the packet. That fulfillment validates the ILP packet sent from the receiver, effectively reducing the balance owed from the sender to the receiver by the amount in the original ILP packet (the payment request).
+
+The flow goes as follows:
+
+1. The receiver creates the payment request, an `InterledgerPrepare` packet, with:
+    1.  the `amount` that is being paid from the sender to the receiver (in the receiver's currency and scale),
+    2.  a `destination` ILP Address that routes to the receiver (and may contain a payment specific suffix to help the receiver identify it when it arrives),
+    3.  an `executionCondition` generated using one of algorithms outlined in [Condition Generation Recommendations](#condition-generation-recommendations),
+    4.  an appropriate `expiresAt` that will allow the sender to both, determine the appropriate amount to send (a process that could potentially take some time), and complete the payment, and
+    5.  optional application `data` that is opaque to the sender.
+2. The receiver sends the payment request (the `InterledgerPrepare` packet) to the sender.
+3. The sender routes this packet out via one of its ILP peers, behaving like a typical ILP connector, and MUST therefor:
+    1.  determine the appropriate `amount` to send on the next hop,
```

Comment:
How does the sender determine the rate between their units and the receiver's units? As @michielbdejong mentioned, IPRv1 depended on ILQP, but since that was removed from the Interledger layer transport protocols need to specify a quoting mechanism.

Also, what happens if the amount the receiver requests is larger than the Maximum Packet Amount for the path from the sender to the receiver? The receiver has no idea what the limit might be, because they don't know where the sender is sending from

Reply:
> How does the sender determine the rate between their units and the receiver's units? As @michielbdejong mentioned, IPRv1 depended on ILQP, but since that was removed from the Interledger layer 

That's not in scope for this protocol. The sender can use whatever mechanism they wish, they could send a probing payment, query a third-party or send an amount hey consider "enough"

> transport protocols need to specify a quoting mechanism.

Why? How you determine the amount to send shouldn't be tightly coupled to the transport protocol.